### DISCUSSION TITLE
Is there a way to change the environment for a branch-domain (.vercel.app)?

### DISCUSSION THREAD

**Discussion by julietgar** [vercel/community#2965] [Help]
### Summary

I'm trying to find a way to set a different environment variables for the branches pushed to git.

### Steps to Reproduce

Hi! 

I'm trying to find a way to set different environment variables for the branches pushed to git.

I would like that for a particular branch: `123-example-branch` I can set an specific environment for all of them. an environment isolate to the rest. I have some env variable set for the "preview" environment but it just can work for staging branch changes, these other variables going to have other values and pointing to another backend environment.

I put here what I try to do:

![Screen Shot 2023-06-22 at 23 35 26](https://github.com/vercel/community/assets/2846595/c0214981-8dfd-4457-9fb7-083fea8eb15e)

In "Environment" I would like change it to choose other.

I add more details:

Nowadays I have these environments working: production and preview, they have a specific branch set:

- production with main branch
- preview with staging branch

I would like that other particular branches pushed, for example, `123-branch-name` or similiar different than `main` and `staging`, can have set a different environment, for example, `beta` but I see that always they have `preview` environment set. How can I change that?

I see that Vercel generates subdomains with some hashes by branch, for example, admin-abc1234hjg-trady.vercel.app and they have an environment set by default that is `preview` but I would like to change that to set a new environment.

Thanks! 🙏

**Comment by smaeda-ks** [comment]
@julietgar you can set preview environment variables to a specific git branch:
https://vercel.com/docs/concepts/projects/environment-variables#preview-environment-variables

Otherwise, could you give some more specific examples?

**Reply by julietgar** [reply]
Yes, nowadays I have these environments working: production and preview, they have a specific branch set:

- production with main branch
- preview with staging branch

I would like that other particular branches pushed, for example, `123-branch-name` or similiar different than `main` and `staging`, can have set a different environment, for example, `beta` but I see that always they have `preview` environment set. How can I change that?

I see that Vercel generates subdomains with some hashes by branch, for example, admin-abc1234hjg-trady.vercel.app and they have an environment set by default that is `preview` but I would like to change that to set a new environment.

**Reply by smaeda-ks** [reply]
@julietgar Thanks. Now it's getting more clear. But what is the main reason for having different environments? Would like to know more about your exact use case.

On Vercel, we only have two environments, which are `production` and `preview`, nothing else. There's no way to add additional environments, and we don't have any plan to support that.

**Reply by julietgar** [reply]
Yes, the use case is, I want to do this to test specific changes from a specific branch, for example, I worked changing the user login logic in backend, I have those changes in this branch `123-user-login` but nowadays to test them "publicly", I have to merge to `staging` branch. 

On Vercel are the frontend changes in a branch called `456-user-login` but to test them, also I have to merge to `staging` branch, so I want an isolate environment for these feature branches. In backend, I solved this creating a new environment called `beta` where feature branch changes can be pushed there without to merge to `staging` branch, I configured this with github actions/workflows but now I need to point the frontend feature branches in vercel to the `beta` env because are pointing to `staging` environtment so, I would like to have a new environment variables to do that.

Some suggestion to solve this?
Thanks!

**Reply by julietgar** [reply]
Hi @smaeda-ks do you have some suggestion for my case?

Thanks!
